server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -450,6 +437,9 @@ synchronized void updateRemoteCluster(
             connectionMap.put(clusterAlias, remote);
             remote.ensureConnected(listener.map(ignored -> RemoteClusterConnectionStatus.RECONNECTED));
         } else {
+            if (crossProjectEnabled == false) {
+                remote.setSkipUnavailable(config.skipUnavailable());
+            }
             // No changes to connection configuration.
             listener.onResponse(RemoteClusterConnectionStatus.UNCHANGED);
```

Comment:
It is a little odd that we report `UNCHANGED` when `skipUnavailable` changes. It might be useful to add a new enum such as `UPDATED` for updating properties on existing connection.